lib/node_modules/@stdlib/random/base/mt19937/test/test.factory.js

```diff
@@ -0,0 +1,269 @@
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof factory, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'if provided a value which is not a positive integer, the factory function throws an error', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		3.14,
+		0.0,
+		-5.0,
+		NaN,
+		true,
+		null,
+		undefined,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			factory( value );
+		};
+	}
+});
+
+tape( 'the function throws a range error if provided an integer greater than or equal to the maximum signed 32-bit integer', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		INT32_MAX,
+		INT32_MAX + 1,
+		INT32_MAX + 2
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), RangeError, 'throws a range error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			factory( value );
+		};
+	}
+});
+
+tape( 'the function returns a pseudorandom number generator (no seed)', function test( t ) {
+	var mt19937;
+	var v;
+	var i;
+
+	mt19937 = factory();
+	for ( i = 0; i < 1e4; i++ ) {
+		v = mt19937();
+		t.equal( typeof v, 'number', 'returns a number' );
+		t.equal( isPositiveInteger( v ), true, 'returns a positive integer' );
+		t.equal( v >= 1 && v <= UINT32_MAX-1, true, 'returns an integer between 1 and 2^31-1 (inclusive)' );
+	}
+	t.end();
+});
+
+tape( 'the function returns a seeded pseudorandom number generator', function test( t ) {
+	var mt199371;
+	var mt199372;
+	var seed;
+	var v1;
+	var v2;
+	var i;
+
+	seed = round( Date.now() / 1000 );
+
+	mt199371 = factory( seed );
+	mt199372 = factory( seed );
+
+	t.notEqual( mt199371, mt199372, 'separate generators' );
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v1 = mt199371();
+		v2 = mt199372();
+		t.equal( v1, v2, 'both return same number' );
+	}
+	t.end();
+});
+
+tape( 'attached to the returned function is the generator name', function test( t ) {
+	var mt19937 = factory();
+	t.equal( mt19937.NAME, 'mt19937', 'has `NAME` property' );
+	t.end();
+});
+
+tape( 'attached to the returned function is the generator seed', function test( t ) {
+	var mt19937 = factory( 12345 );
+	t.equal( typeof mt19937.SEED, 'number', 'has `SEED` property' );
+	t.equal( mt19937.SEED, 12345, 'equal to provided seed' );
+	t.end();
+});
+
+tape( 'attached to the returned function is the minimum possible generated number', function test( t ) {
+	var mt19937 = factory();
+	t.equal( mt19937.MIN, 1, 'has `MIN` property' );
+	t.end();
+});
+
+tape( 'attached to the returned function is the maximum possible generated number', function test( t ) {
+	var mt19937 = factory();
+	t.equal( mt19937.MAX, UINT32_MAX-1, 'has `MAX` property' );
```

Comment:
Why is the max `UINT32_MAX-1`? According to the C source, I might expect `UINT32_MAX`.

```
UINT32_MAX == 0xffffffff
```